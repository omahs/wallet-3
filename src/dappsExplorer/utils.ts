import { DappV1, DappV2, isDappV2 } from 'src/dapps/types'

export const calculateSearchScore = (dapp: DappV2 | DappV1, searchQuery: string) => {
  const nameMatchingCaseScore = dapp.name.includes(searchQuery) ? 2 : 0
  const nameScore = dapp.name.toLowerCase().includes(searchQuery.toLowerCase()) ? 1 : 0
  const descriptionScore = dapp.description.toLowerCase().includes(searchQuery.toLowerCase())
    ? 1
    : 0
  const categoryScore = isDappV2(dapp)
    ? dapp.categories.some((category) => category.toLowerCase().includes(searchQuery.toLowerCase()))
      ? 1
      : 0
    : dapp.categoryId.toLowerCase().includes(searchQuery.toLowerCase())
    ? 1
    : 0
  return nameMatchingCaseScore + nameScore + descriptionScore + categoryScore
}
