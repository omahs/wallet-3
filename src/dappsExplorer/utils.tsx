import { DappV1, DappV2, isDappV2 } from 'src/dapps/types'

/***
 * Calculates a score based on the search term and dapp
 * - +2 if the name matches the search term in a case sensitive way
 * - +1 if the name matches the search term in a case insensitive way
 * - +1 if the description matches the search term in a case insensitive way
 * - +1 if the category matches the search term in a case insensitive way
 * @param dapp
 * @param searchTerm
 * @returns {number} the score of the dapp 0 - 5
 */
export const calculateSearchScore = (dapp: DappV2 | DappV1, searchTerm: string) => {
  if (searchTerm === '') return 0
  const nameMatchingCaseScore = dapp.name.includes(searchTerm) ? 2 : 0
  const nameScore = dapp.name.toLowerCase().includes(searchTerm.toLowerCase()) ? 1 : 0
  const descriptionScore = dapp.description.toLowerCase().includes(searchTerm.toLowerCase()) ? 1 : 0
  const categoryScore = isDappV2(dapp)
    ? dapp.categories.some((category) => category.toLowerCase().includes(searchTerm.toLowerCase()))
      ? 1
      : 0
    : dapp.categoryId.toLowerCase().includes(searchTerm.toLowerCase())
    ? 1
    : 0
  return nameMatchingCaseScore + nameScore + descriptionScore + categoryScore
}
