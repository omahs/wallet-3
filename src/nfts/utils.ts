import Share from 'react-native-share'
import { Nft } from 'src/nfts/types'
import { fetchImageAsBase64, saveRemoteToCameraRoll } from 'src/utils/image'
import Logger from 'src/utils/Logger'

const TAG = 'nfts/utils'

// TODO: Add analytics for sharing nft
export const onShare = async (nft: Nft) => {
  try {
    const url =
      nft.media.find((media) => media.raw === nft.metadata?.image)?.gateway ?? nft.metadata?.image
    if (!url) throw new Error('No image url found')
    const { data, mimeType } = await fetchImageAsBase64(url)
    const dataUrl = `data:${mimeType};base64,${data}`
    return Share.open({
      url: dataUrl,
      type: mimeType,
      failOnCancel: false, // don't throw if user cancels share
    })
  } catch (error) {
    Logger.error(TAG, 'Error sharing nft', error)
  }
}

export const onSave = async (nft: Nft) => {
  try {
    const url =
      nft.media.find((media) => media.raw === nft.metadata?.image)?.gateway ?? nft.metadata?.image
    if (!url) throw new Error('No image url found')
    await saveRemoteToCameraRoll(url)
  } catch (error) {
    Logger.error(TAG, 'Error saving nft', error)
  }
}
