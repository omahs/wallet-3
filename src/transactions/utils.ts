import { CeloTxReceipt } from '@celo/connect'
import BigNumber from 'bignumber.js'
import i18n from 'src/i18n'
import { tokensByIdSelector } from 'src/tokens/selectors'
import { getTokenId } from 'src/tokens/utils'
import { NetworkId, TokenTransaction, TransactionStatus } from 'src/transactions/types'
import Logger from 'src/utils/Logger'
import { formatFeedSectionTitle, timeDeltaInDays } from 'src/utils/time'
import { select } from 'typed-redux-saga'
import { TransactionReceipt } from 'viem'

// Groupings:
// Recent -> Last 7 days (pending transactions always at the top, followed by recent confirmed transactions).
// [Current month] - "July" -> Captures transactions from the current month that aren’t captured in Recent.
// [Previous months] - "June" -> Captures transactions by month.
// [Months over a year ago] — "July 2019" -> Same as above, but with year appended.
// Sections are hidden if they have no items.
export function groupFeedItemsInSections(
  pendingTransactions: TokenTransaction[],
  confirmedTransactions: TokenTransaction[]
) {
  const sectionsMap: {
    [key: string]: {
      data: TokenTransaction[]
      daysSinceTransaction: number
    }
  } = {}

  // add standby transactions to top of recent section
  const recentSectionTitle = i18n.t('feedSectionHeaderRecent')
  if (pendingTransactions.length > 0) {
    sectionsMap[recentSectionTitle] = {
      daysSinceTransaction: 0,
      data: pendingTransactions,
    }
  }

  confirmedTransactions.forEach((transaction) => {
    const daysSinceTransaction = timeDeltaInDays(Date.now(), transaction.timestamp)
    const sectionTitle =
      daysSinceTransaction <= 7
        ? i18n.t('feedSectionHeaderRecent')
        : formatFeedSectionTitle(transaction.timestamp, i18n)
    sectionsMap[sectionTitle] = {
      daysSinceTransaction: sectionsMap[sectionTitle]?.daysSinceTransaction ?? daysSinceTransaction,
      data: [...(sectionsMap[sectionTitle]?.data ?? []), transaction],
    }
  })

  return Object.entries(sectionsMap)
    .sort((a, b) => a[1].daysSinceTransaction - b[1].daysSinceTransaction)
    .map(([key, value]) => ({
      title: key,
      data: value.data,
    }))
}

export function* buildBaseTransactionReceipt(
  receipt: TransactionReceipt | CeloTxReceipt,
  networkId: NetworkId
) {
  const tokensById = yield* select((state) => tokensByIdSelector(state, [networkId]))

  // Receipt gas information is based on the native token (there is no feeCurrency)
  const feeCurrencyId = getTokenId(networkId)
  const feeTokenInfo = tokensById[feeCurrencyId]

  const gasFeeInWei = new BigNumber(receipt.gasUsed.toString()).times(
    new BigNumber(receipt.effectiveGasPrice.toString())
  )

  if (!feeTokenInfo) {
    Logger.warn(`No information found for token ${feeCurrencyId}`)
  }

  const baseDetails = {
    status:
      receipt.status === 'reverted' || !receipt.status
        ? TransactionStatus.Failed
        : TransactionStatus.Complete,
    block: receipt.blockNumber.toString(),
    transactionHash: receipt.transactionHash,
  }
  const feeDetails = !feeTokenInfo
    ? {}
    : {
        feeCurrencyId: feeTokenInfo.tokenId,
        gasFee: gasFeeInWei.shiftedBy(-feeTokenInfo.decimals).toFixed(),
      }

  return {
    ...baseDetails,
    ...feeDetails,
  }
}
