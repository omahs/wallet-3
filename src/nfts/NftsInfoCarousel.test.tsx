import { fireEvent, render } from '@testing-library/react-native'
import * as React from 'react'
import { navigate, navigateBack } from 'src/navigator/NavigationService'
import { Screens } from 'src/navigator/Screens'
import NftsInfoCarousel from 'src/nfts/NftsInfoCarousel'
import networkConfig from 'src/web3/networkConfig'
import { getMockStackScreenProps } from 'test/utils'
import { mockNft, mockNft2, mockNft3 } from 'test/values'

describe('NftsInfoCarousel', () => {
  it('renders correctly with one Nft', () => {
    const tree = render(
      <NftsInfoCarousel
        {...getMockStackScreenProps(Screens.NftsInfoCarousel, { nfts: [mockNft] })}
      />
    )

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with two valid Nfts', () => {
    const tree = render(
      <NftsInfoCarousel
        {...getMockStackScreenProps(Screens.NftsInfoCarousel, { nfts: [mockNft, mockNft2] })}
      />
    )

    expect(tree).toMatchSnapshot()
  })

  it('renders error state when no Nft(s)', () => {
    const tree = render(
      <NftsInfoCarousel {...getMockStackScreenProps(Screens.NftsInfoCarousel, { nfts: [] })} />
    )
    expect(tree).toMatchSnapshot()
  })

  it('renders error image state on Nft null metadata', () => {
    const tree = render(
      <NftsInfoCarousel
        {...getMockStackScreenProps(Screens.NftsInfoCarousel, { nfts: [mockNft3] })}
      />
    )
    expect(tree).toMatchSnapshot()
  })

  it('image carousel should not render Nfts with null metadata', () => {
    const { queryByTestId } = render(
      <NftsInfoCarousel
        {...getMockStackScreenProps(Screens.NftsInfoCarousel, { nfts: [mockNft, mockNft3] })}
      />
    )

    // Two Nfts but one with null metadata will prevent the image carousel from rendering
    expect(queryByTestId('NftsInfoCarousel/NftImageCarousel')).toBeNull()
  })

  it('toggle active Nft', () => {
    const { getByTestId, getByText } = render(
      <NftsInfoCarousel
        {...getMockStackScreenProps(Screens.NftsInfoCarousel, { nfts: [mockNft, mockNft2] })}
      />
    )

    // Toggle to Second Nft
    fireEvent.press(
      getByTestId(`NftsInfoCarousel/NftThumbnail/${mockNft2.contractAddress}-${mockNft2.tokenId}`)
    )
    expect(getByText(mockNft2.metadata!.name)).toBeTruthy()

    // Return to first Nft
    fireEvent.press(
      getByTestId(`NftsInfoCarousel/NftThumbnail/${mockNft.contractAddress}-${mockNft.tokenId}`)
    )
    expect(getByText(mockNft.metadata!.name)).toBeTruthy()
  })

  it('should be able to navigate back', () => {
    const { getByTestId } = render(
      <NftsInfoCarousel
        {...getMockStackScreenProps(Screens.NftsInfoCarousel, { nfts: [mockNft] })}
      />
    )

    fireEvent.press(getByTestId('NftsInfoCarousel/BackButton'))
    expect(navigateBack).toHaveBeenCalled()
  })

  it('opens link for Explorer', () => {
    const { getByTestId } = render(
      <NftsInfoCarousel
        {...getMockStackScreenProps(Screens.NftsInfoCarousel, { nfts: [mockNft] })}
      />
    )

    fireEvent.press(getByTestId('ViewOnExplorer'))
    expect(navigate).toHaveBeenCalledWith(Screens.WebViewScreen, {
      uri: `${networkConfig.celoExplorerBaseTokenUrl}${mockNft.contractAddress}/instance/${mockNft.tokenId}/metadata`,
    })
  })
})
