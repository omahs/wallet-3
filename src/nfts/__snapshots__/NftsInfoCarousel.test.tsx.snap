// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NftsInfoCarousel renders correctly with one Nft 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  testID="NftsInfoCarousel"
>
  <View
    style={
      Object {
        "flex": 1,
        "zIndex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginHorizontal": 16,
          "marginTop": 16,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#2E3338",
            "borderRadius": 100,
            "elevation": 4,
            "height": 32,
            "justifyContent": "center",
            "shadowColor": "#2E3338",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 4,
            "width": 32,
          }
        }
        testID="NftsInfoCarousel/BackButton"
      >
        <svg
          fill="none"
          height={32}
          viewBox="0 0 32 32"
          width={32}
        >
          <path
            d="M19.4303 9.4303L17.9919 8L10 16L18 24L19.4303 22.5697L12.8606 16L19.4303 9.4303Z"
            fill="#2E3338"
          />
        </svg>
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "height": 360,
                "width": 750,
              },
              Object {
                "borderRadius": 8,
              },
            ],
          ]
        }
      >
        <FastImageView
          defaultSource={null}
          onFastImageError={[Function]}
          onFastImageLoad={[Function]}
          onFastImageLoadEnd={[Function]}
          resizeMode="contain"
          source={
            Object {
              "uri": "https://example.com/1",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
        <View
          onLayout={[Function]}
        >
          <View
            style={
              Object {
                "backgroundColor": "transparent",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "height": 360,
                    "width": 750,
                    "zIndex": -1,
                  },
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Object {
                      "backgroundColor": "#E6E6E6",
                    },
                  ],
                  Object {
                    "borderRadius": 8,
                    "height": 360,
                    "width": 750,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginHorizontal": 24,
            "marginTop": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2E3338",
              "fontFamily": "Jost-Book",
              "fontSize": 26,
              "lineHeight": 32,
            }
          }
        >
          John Doe.fizzBuzz
        </Text>
      </View>
      <View
        style={
          Object {
            "marginHorizontal": 24,
            "marginTop": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2E3338",
              "fontFamily": "Inter-SemiBold",
              "fontSize": 18,
              "lineHeight": 24,
              "marginBottom": 16,
            }
          }
        >
          nftInfoCarousel.description
        </Text>
        <Text
          style={
            Object {
              "color": "#2E3338",
              "fontFamily": "Inter-Regular",
              "fontSize": 16,
              "lineHeight": 22,
            }
          }
        >
          This is a fizzBuzz name!
        </Text>
      </View>
      <View
        style={
          Object {
            "marginHorizontal": 24,
            "marginTop": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2E3338",
              "fontFamily": "Inter-SemiBold",
              "fontSize": 18,
              "lineHeight": 24,
              "marginBottom": 16,
            }
          }
        >
          nftInfoCarousel.attributes
        </Text>
        <View
          style={
            Object {
              "paddingBottom": 24,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#757575",
                "fontFamily": "Inter-Medium",
                "fontSize": 14,
                "lineHeight": 18,
              }
            }
          >
            Fizz Buzz
          </Text>
          <Text
            style={
              Object {
                "color": "#2E3338",
                "fontFamily": "Inter-Regular",
                "fontSize": 16,
                "lineHeight": 22,
              }
            }
          >
            1
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginHorizontal": 24,
            "marginTop": 16,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          nativeBackgroundAndroid={
            Object {
              "attribute": "selectableItemBackground",
              "rippleRadius": undefined,
              "type": "ThemeAttrAndroid",
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="ViewOnExplorer"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#178154",
                  "fontFamily": "Inter-Medium",
                  "fontSize": 14,
                  "lineHeight": 18,
                  "paddingRight": 8,
                }
              }
            >
              nftInfoCarousel.viewOnCeloExplorer
            </Text>
            <svg
              fill="none"
              height={12}
              width={12}
            >
              <path
                d="M10.148 10.148H1.852V1.852H6V.667H1.852C1.194.667.667 1.2.667 1.852v8.296c0 .652.527 1.185 1.185 1.185h8.296c.652 0 1.185-.533 1.185-1.185V6h-1.185v4.148ZM7.185.667v1.185h2.127L3.487 7.677l.836.836 5.825-5.826v2.128h1.185V.667H7.185Z"
                fill="#178154"
              />
            </svg>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;

exports[`NftsInfoCarousel renders correctly with two valid Nfts 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  testID="NftsInfoCarousel"
>
  <View
    style={
      Object {
        "flex": 1,
        "zIndex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginHorizontal": 16,
          "marginTop": 16,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#2E3338",
            "borderRadius": 100,
            "elevation": 4,
            "height": 32,
            "justifyContent": "center",
            "shadowColor": "#2E3338",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 4,
            "width": 32,
          }
        }
        testID="NftsInfoCarousel/BackButton"
      >
        <svg
          fill="none"
          height={32}
          viewBox="0 0 32 32"
          width={32}
        >
          <path
            d="M19.4303 9.4303L17.9919 8L10 16L18 24L19.4303 22.5697L12.8606 16L19.4303 9.4303Z"
            fill="#2E3338"
          />
        </svg>
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "height": 360,
                "width": 750,
              },
              Object {
                "borderRadius": 8,
              },
            ],
          ]
        }
      >
        <FastImageView
          defaultSource={null}
          onFastImageError={[Function]}
          onFastImageLoad={[Function]}
          onFastImageLoadEnd={[Function]}
          resizeMode="contain"
          source={
            Object {
              "uri": "https://example.com/1",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
        <View
          onLayout={[Function]}
        >
          <View
            style={
              Object {
                "backgroundColor": "transparent",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "height": 360,
                    "width": 750,
                    "zIndex": -1,
                  },
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Object {
                      "backgroundColor": "#E6E6E6",
                    },
                  ],
                  Object {
                    "borderRadius": 8,
                    "height": 360,
                    "width": 750,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RCTScrollView
          contentContainerStyle={
            Object {
              "alignItems": "center",
              "flexGrow": 1,
              "justifyContent": "center",
              "marginLeft": 16,
            }
          }
          horizontal={true}
          showsHorizontalScrollIndicator={false}
          style={
            Object {
              "flexDirection": "row",
              "paddingHorizontal": 16,
              "paddingTop": 8,
            }
          }
          testID="NftsInfoCarousel/NftImageCarousel"
        >
          <View>
            <View
              style={
                Object {
                  "borderRadius": 8,
                  "marginRight": 8,
                  "overflow": "hidden",
                }
              }
              testID="NftsInfoCarousel/NftThumbnail/0x000000000000000000000000000000000000CE10-1"
            >
              <View
                accessible={true}
                focusable={true}
                nativeBackgroundAndroid={
                  Object {
                    "attribute": "selectableItemBackground",
                    "rippleRadius": undefined,
                    "type": "ThemeAttrAndroid",
                  }
                }
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
              >
                <View
                  style={
                    Array [
                      Object {
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "borderRadius": 8,
                        },
                        Object {
                          "height": 40,
                          "width": 40,
                        },
                      ],
                    ]
                  }
                >
                  <FastImageView
                    defaultSource={null}
                    onFastImageError={[Function]}
                    onFastImageLoadEnd={[Function]}
                    resizeMode="cover"
                    source={
                      Object {
                        "uri": "https://example.com/1",
                      }
                    }
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  />
                  <View
                    onLayout={[Function]}
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "transparent",
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "height": 40,
                              "width": 40,
                              "zIndex": -1,
                            },
                            Array [
                              Object {
                                "overflow": "hidden",
                              },
                              Object {
                                "backgroundColor": "#E6E6E6",
                              },
                            ],
                            Object {
                              "borderRadius": undefined,
                              "height": 40,
                              "width": 40,
                            },
                          ]
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "borderRadius": 8,
                  "marginRight": 8,
                  "overflow": "hidden",
                }
              }
              testID="NftsInfoCarousel/NftThumbnail/0x000000000000000000000000000000000000CE10-3"
            >
              <View
                accessible={true}
                focusable={true}
                nativeBackgroundAndroid={
                  Object {
                    "attribute": "selectableItemBackground",
                    "rippleRadius": undefined,
                    "type": "ThemeAttrAndroid",
                  }
                }
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
              >
                <View
                  style={
                    Array [
                      Object {
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "borderRadius": 8,
                        },
                        Object {
                          "height": 32,
                          "opacity": 0.5,
                          "width": 32,
                        },
                      ],
                    ]
                  }
                >
                  <FastImageView
                    defaultSource={null}
                    onFastImageError={[Function]}
                    onFastImageLoadEnd={[Function]}
                    resizeMode="cover"
                    source={
                      Object {
                        "uri": "https://example.com/3",
                      }
                    }
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  />
                  <View
                    onLayout={[Function]}
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "transparent",
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "height": 40,
                              "width": 40,
                              "zIndex": -1,
                            },
                            Array [
                              Object {
                                "overflow": "hidden",
                              },
                              Object {
                                "backgroundColor": "#E6E6E6",
                              },
                            ],
                            Object {
                              "borderRadius": undefined,
                              "height": 40,
                              "width": 40,
                            },
                          ]
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
      <View
        style={
          Object {
            "marginHorizontal": 24,
            "marginTop": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2E3338",
              "fontFamily": "Jost-Book",
              "fontSize": 26,
              "lineHeight": 32,
            }
          }
        >
          John Doe.fizzBuzz
        </Text>
      </View>
      <View
        style={
          Object {
            "marginHorizontal": 24,
            "marginTop": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2E3338",
              "fontFamily": "Inter-SemiBold",
              "fontSize": 18,
              "lineHeight": 24,
              "marginBottom": 16,
            }
          }
        >
          nftInfoCarousel.description
        </Text>
        <Text
          style={
            Object {
              "color": "#2E3338",
              "fontFamily": "Inter-Regular",
              "fontSize": 16,
              "lineHeight": 22,
            }
          }
        >
          This is a fizzBuzz name!
        </Text>
      </View>
      <View
        style={
          Object {
            "marginHorizontal": 24,
            "marginTop": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2E3338",
              "fontFamily": "Inter-SemiBold",
              "fontSize": 18,
              "lineHeight": 24,
              "marginBottom": 16,
            }
          }
        >
          nftInfoCarousel.attributes
        </Text>
        <View
          style={
            Object {
              "paddingBottom": 24,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#757575",
                "fontFamily": "Inter-Medium",
                "fontSize": 14,
                "lineHeight": 18,
              }
            }
          >
            Fizz Buzz
          </Text>
          <Text
            style={
              Object {
                "color": "#2E3338",
                "fontFamily": "Inter-Regular",
                "fontSize": 16,
                "lineHeight": 22,
              }
            }
          >
            1
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginHorizontal": 24,
            "marginTop": 16,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          nativeBackgroundAndroid={
            Object {
              "attribute": "selectableItemBackground",
              "rippleRadius": undefined,
              "type": "ThemeAttrAndroid",
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="ViewOnExplorer"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#178154",
                  "fontFamily": "Inter-Medium",
                  "fontSize": 14,
                  "lineHeight": 18,
                  "paddingRight": 8,
                }
              }
            >
              nftInfoCarousel.viewOnCeloExplorer
            </Text>
            <svg
              fill="none"
              height={12}
              width={12}
            >
              <path
                d="M10.148 10.148H1.852V1.852H6V.667H1.852C1.194.667.667 1.2.667 1.852v8.296c0 .652.527 1.185 1.185 1.185h8.296c.652 0 1.185-.533 1.185-1.185V6h-1.185v4.148ZM7.185.667v1.185h2.127L3.487 7.677l.836.836 5.825-5.826v2.128h1.185V.667H7.185Z"
                fill="#178154"
              />
            </svg>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;

exports[`NftsInfoCarousel renders error image state on Nft null metadata 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  testID="NftsInfoCarousel"
>
  <View
    style={
      Object {
        "flex": 1,
        "zIndex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginHorizontal": 16,
          "marginTop": 16,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#2E3338",
            "borderRadius": 100,
            "elevation": 4,
            "height": 32,
            "justifyContent": "center",
            "shadowColor": "#2E3338",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 4,
            "width": 32,
          }
        }
        testID="NftsInfoCarousel/BackButton"
      >
        <svg
          fill="none"
          height={32}
          viewBox="0 0 32 32"
          width={32}
        >
          <path
            d="M19.4303 9.4303L17.9919 8L10 16L18 24L19.4303 22.5697L12.8606 16L19.4303 9.4303Z"
            fill="#2E3338"
          />
        </svg>
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2E3338",
            "height": 360,
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
            }
          }
        >
          nftInfoCarousel.nftImageLoadError
        </Text>
      </View>
      <View
        style={
          Object {
            "marginHorizontal": 24,
            "marginTop": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2E3338",
              "fontFamily": "Jost-Book",
              "fontSize": 26,
              "lineHeight": 32,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "marginHorizontal": 24,
            "marginTop": 16,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          nativeBackgroundAndroid={
            Object {
              "attribute": "selectableItemBackground",
              "rippleRadius": undefined,
              "type": "ThemeAttrAndroid",
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="ViewOnExplorer"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#178154",
                  "fontFamily": "Inter-Medium",
                  "fontSize": 14,
                  "lineHeight": 18,
                  "paddingRight": 8,
                }
              }
            >
              nftInfoCarousel.viewOnCeloExplorer
            </Text>
            <svg
              fill="none"
              height={12}
              width={12}
            >
              <path
                d="M10.148 10.148H1.852V1.852H6V.667H1.852C1.194.667.667 1.2.667 1.852v8.296c0 .652.527 1.185 1.185 1.185h8.296c.652 0 1.185-.533 1.185-1.185V6h-1.185v4.148ZM7.185.667v1.185h2.127L3.487 7.677l.836.836 5.825-5.826v2.128h1.185V.667H7.185Z"
                fill="#178154"
              />
            </svg>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;

exports[`NftsInfoCarousel renders error state when no Nft(s) 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="NftsInfoCarousel/NftsLoadErrorScreen"
>
  <Text>
    nftInfoCarousel.noNftsFound
  </Text>
</RNCSafeAreaView>
`;
